src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs

```diff
@@ -45,7 +50,25 @@ public static ProjectInfo CreateProjectInfo(this ProjectFileInfo projectFileInfo
                 language: LanguageNames.CSharp,
                 filePath: projectFileInfo.FilePath,
                 outputFilePath: projectFileInfo.TargetPath,
-                compilationOptions: projectFileInfo.CreateCompilationOptions());
+                compilationOptions: projectFileInfo.CreateCompilationOptions(),
+                analyzerReferences: analyzerReferences);
+        }
+
+        private static IEnumerable<AnalyzerReference> ResolveAnalyzerReferencesForProject(ProjectFileInfo projectFileInfo)
+        {
+            return projectFileInfo.Analyzers
+                .GroupBy(x => Path.GetDirectoryName(x))
+                .SelectMany(singleAnalyzerPackageGroup =>
+                {
+                    // Is there better way to figure out entry assembly for specific nuget analyzer package? And is there even entry assembly or is better way to just lookup all referenced assemblies?
+                    return singleAnalyzerPackageGroup
+                        .Where(x => x.EndsWith("Analyzers.dll") || x.EndsWith("Analyzer.dll"))
+                        .Select(analyzerMainAssembly => {
+                            var assemblyLoader = new AnalyzerAssemblyLoader();
+                            singleAnalyzerPackageGroup.ToList().ForEach(x => assemblyLoader.AddDependencyLocation(x));
```

Comment:
`To List()` requires a heap allocation and it's just to perform a foreach. Maybe just use a foreach statement? :smile: